### ISSUE TITLE
the question about the loss function in the paper

### ISSUE THREAD

**Issue by Snmoflower** [ando-khachatryan/HiDDeN#34]
Regarding the optimization target L_G for the encoder, I think it should optimize log(A(I_en)) instead of log(1- A(I_en)). This is because the generator needs to generate targets that the discriminator cannot distinguish, meaning the value of A(I_en) should decrease, and consequently, the value of log should also decrease. It seems like the loss function in the original text is written in the opposite way. Can someone help me solve this question?

**Comment by ando-khachatryan** [comment]
Yep, there is a mistake in the paper with one of the GAN-related loss terms, it should be flipped. The code in the repo is correct.

**Comment by Snmoflower** [comment]
Thank you for replying. But for the code in the repo, i see
`d_loss_on_encoded = self.bce_with_logits_loss(d_on_encoded, d_target_label_encoded)`
`g_loss_adv = self.bce_with_logits_loss(d_on_encoded_for_enc, g_target_label_encoded)`
in the code of training the discriminater and the generater. 
When training the discriminator, it makes sense for the loss to decrease, as the discriminator gets better at distinguishing images. However, when training the generator, the opposite should happen. In other words, the generator should generate images that lead to an increase in the same loss, making it more challenging for the discriminator to distinguish the generated images. In the following GAN code, we can observe that the loss is set in the opposite way during the training of the discriminator and the generator.
`fake_loss = adversarial_loss(discriminator(gen_imgs.detach()), fake)`
`g_loss = adversarial_loss(discriminator(gen_imgs), valid)`
[https://github.com/eriklindernoren/PyTorch-GAN/blob/master/implementations/gan/gan.py](url)
Do I misunderstand?

**Comment by ando-khachatryan** [comment]
Have you seen how d_target_label_encoded and g_target_label_encoded are defined? They are the opposite of each other.

**Comment by Snmoflower** [comment]
Oh, yes, it was my oversight. Thank you for your patient explanation.
